src/main/java/com/faforever/client/teammatchmaking/TeamMatchmakingController.java

```diff
@@ -0,0 +1,324 @@
+
+  private final CountryFlagService countryFlagService;
+  private final AvatarService avatarService;
+  private final PlayerService playerService;
+  private final I18n i18n;
+  private final UiService uiService;
+  private final TeamMatchmakingService teamMatchmakingService;
+  private final FafService fafService;
+  private final EventBus eventBus;
+
+  public StackPane teamMatchmakingRoot;
+  public Button invitePlayerButton;
+  public Button leavePartyButton;
+  public Label refreshingLabel;
+  public ToggleButton uefButton;
+  public ToggleButton cybranButton;
+  public ToggleButton aeonButton;
+  public ToggleButton seraphimButton;
+  public ImageView avatarImageView;
+  public ImageView countryImageView;
+  public Label clanLabel;
+  public Label usernameLabel;
+  public Label gameCountLabel;
+  public Label leagueLabel;
+  public HBox queueBox;
+  public GridPane partyMemberPane;
+  public VBox preparationArea;
+  public ImageView leagueImageView;
+  public Label matchmakerHeadingLabel;
+  public Label partyHeadingLabel;
+  public Label queueHeadingLabel;
+  public ScrollPane scrollPane;
+  public HBox playerCard;
+  public Label crownLabel;
+  public TabPane chatTabPane;
+  public GridPane contentPane;
+  public ColumnConstraints column2;
+  public RowConstraints row2;
+  private Player player;
+  @VisibleForTesting
+  protected MatchmakingChatController matchmakingChatController;
+
+  @Override
+  public void initialize() {
+    eventBus.register(this);
+    JavaFxUtil.fixScrollSpeed(scrollPane);
+    initializeDynamicChatPosition();
+    player = playerService.getCurrentPlayer().get();
+    initializeUppercaseText();
+    initializeBindings();
+
+    teamMatchmakingService.getParty().getMembers().addListener((Observable o) -> renderPartyMembers());
+    if (teamMatchmakingService.isQueuesReadyForUpdate()) {
+      renderQueues(); // The teamMatchmakingService may already have all queues collected
+    }                 // so we won't get any updates on the following change listener
+    teamMatchmakingService.queuesReadyForUpdateProperty().addListener((observable, oldValue, newValue) -> {
+      if (newValue) {
+        renderQueues();
+      }
+    });
+
+    player.statusProperty().addListener((observable, oldValue, newValue) -> {
+      if (newValue != PlayerStatus.IDLE) {
+        teamMatchmakingService.getPlayersInGame().add(player);
+      } else {
+        teamMatchmakingService.getPlayersInGame().remove(player);
+      }
+    });
+
+    teamMatchmakingService.getParty().getMembers().addListener((InvalidationListener) c -> {
+      refreshingLabel.setVisible(false);
+      selectFactionsBasedOnParty();
+    });
+
+    JavaFxUtil.addListener(teamMatchmakingService.getParty().ownerProperty(), (observable, oldValue, newValue) -> {
+      if (matchmakingChatController != null) {
+        matchmakingChatController.closeChannel();
+      }
+      createChannelTab(String.format("#%s'sParty", newValue.getUsername()));
```

Comment:
This should be a constant as well ideally related to the one used for checking the channel name